File: Departamento/Inventory_Service/app/schemas/inventory.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime

class SupplyItem(BaseModel):
    """
    Model representing an inventory supply item (Legacy ingestion endpoint format).
    """
    id: str = Field(..., description="Unique identifier for the supply item")
    name: str = Field(..., description="Name of the supply item (e.g., 'Syringes', 'Bandages')")
    category: str = Field(..., description="Category of the supply")
    quantity: int = Field(..., ge=0, description="Current quantity in stock")
    threshold: int = Field(default=10, description="Minimum quantity threshold before alerting")
    department_id: str = Field(..., description="ID of the department holding this inventory")
    last_updated: Optional[datetime] = Field(default_factory=datetime.utcnow, description="Last update timestamp")

class IngestionResponse(BaseModel):
    """
    Model for the response of the legacy ingestion endpoint.
    """
    message: str
    item_id: str
    status: str

class StockModificationRequest(BaseModel):
    """
    Model for modifying stock quantity in a department.
    """
    supply_id: int
    quantity_change: float = Field(..., description="Positive for addition, negative for subtraction")
    observations: Optional[str] = None
    user_id: int = Field(..., description="User performing the modification")

class MovementRequest(BaseModel):
    """
    Model for registering a movement between departments.
    """
    supply_id: int
    quantity: float = Field(..., gt=0, description="Quantity to move")
    origin_department_id: int
    destination_department_id: int
    observations: Optional[str] = None
    user_id: int = Field(..., description="User performing the movement")

class OperationResponse(BaseModel):
    """
    Generic response model for operations like stock modifications and movements.
    """
    message: str
    status: str
    movement_id: Optional[int] = None

File: Departamento/Inventory_Service/internal/schemas/inventory.go
package schemas

import "time"

type SupplyItem struct {
    ID           string    `json:"id" binding:"required"`
    Name         string    `json:"name" binding:"required"`
    Category     string    `json:"category" binding:"required"`
    Quantity     int       `json:"quantity" binding:"required,gte=0"`
    Threshold    int       `json:"threshold"`
    DepartmentID string    `json:"department_id" binding:"required"`
    LastUpdated  time.Time `json:"last_updated"`
}

type IngestionResponse struct {
    Message string `json:"message"`
    ItemID  string `json:"item_id"`
    Status  string `json:"status"`
}

type StockModificationRequest struct {
    SupplyID       uint    `json:"supply_id" binding:"required"`
    QuantityChange float64 `json:"quantity_change" binding:"required"`
    Observations   string  `json:"observations"`
    UserID         uint    `json:"user_id" binding:"required"`
}

type MovementRequest struct {
    SupplyID                uint    `json:"supply_id" binding:"required"`
    Quantity                float64 `json:"quantity" binding:"required,gt=0"`
    OriginDepartmentID      uint    `json:"origin_department_id" binding:"required"`
    DestinationDepartmentID uint    `json:"destination_department_id" binding:"required"`
    Observations            string  `json:"observations"`
    UserID                  uint    `json:"user_id" binding:"required"`
}

type OperationResponse struct {
    Message    string `json:"message"`
    Status     string `json:"status"`
    MovementID *uint  `json:"movement_id,omitempty"`
}
type CreateSupplyRequest struct {
	InternalCode  string  `json:"internal_code" binding:"required"`
	Name          string  `json:"name" binding:"required"`
	Description   string  `json:"description"`
	UnitOfMeasure string  `json:"unit_of_measure"`
	MinimumStock  float64 `json:"minimum_stock" binding:"required"`
	CategoryID    uint    `json:"category_id"`
	IsActive       bool    `json:"is_active"`
}

type UpdateSupplyRequest struct {
	InternalCode  string  `json:"internal_code"`
	Name          string  `json:"name"`
	Description   string  `json:"description"`
	UnitOfMeasure string  `json:"unit_of_measure"`
	MinimumStock  float64 `json:"minimum_stock"`
	CategoryID    uint    `json:"category_id"`
	IsActive       *bool   `json:"is_active"`
}
